source/developers-guide/shopware-5-core-service-extensions/index.md

```diff
@@ -16,108 +16,125 @@ menu_order: 130
 The new bundle services like __StoreFrontBundle__, __SearchBundle__ or __SearchBundleDBAL__
 don't contain events or hooks, but can still be replaced or extended.
 
-All services are defined in Shopware's dependency injection container, also known as DI container or DIC. A plugin can replace these services in the DIC:
+All services are defined in Shopware's dependency injection container, also known as DI container or DIC. A plugin (`SwagExample` in this case) can replace these services in the DIC by overriding them in its `services.xml` file:
 
-```php
-public function onDispatchEventListener()
-{
-    $newService = new ReplacementServiceImplementation();
-    Shopware()->Container()->set('shopware_storefront.list_product_service', $newService);
-}
+```xml
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="shopware_storefront.list_product_service" class="SwagExample\Bundle\StoreFrontBundle\ListProductService" />
```

Comment:
did you tested this? I'm not sure, but afaik you also have to use the decorate functionality to replace a service